chrome/content/zotero/xpcom/itemTreeView.js

```diff
@@ -3075,6 +3075,37 @@ Zotero.ItemTreeView.prototype.drop = Zotero.Promise.coroutine(function* (row, or
 	}
 });
 
+// Prompt about invalid drag item on unsuccessful drag
+Zotero.ItemTreeView.prototype.onDragEnd = function(event) {
+	if(event.dataTransfer.dropEffect != "none") {
+		return;
+	}
+		
+	var target = event.target;
+	if (target.tagName != 'treechildren') {
+		return;
+	}
+	
+	event = Zotero.DragDrop.currentEvent;
+	var tree = event.target.parentNode;
+	let row = {}, col = {}, obj = {};
+	tree.treeBoxObject.getCellAt(event.clientX, event.clientY, row, col, obj);
+	var view = tree.ownerDocument.defaultView.ZoteroPane.collectionsView;
+	var treeRow = view.getRow(row.value);
+	
+	// Zotero items in My Pubs only
+	if(tree.id == 'zotero-collections-tree' && treeRow.isPublications()) {
+		// Due to a really bizzare bug, the alert needs to be displayed async
+		// https://bugzilla.mozilla.org/show_bug.cgi?id=100180
+		setTimeout(function() { 
+			Services.prompt.alert(null,
+			Zotero.getString('drag.invalid.title'),
+			Zotero.getString('drag.publications.onlyTopLevel')) 
+		}, 50);
+	}
+	
+}
+
```

Comment:
Sorry, not sure I'm following. I wasn't suggesting that you need `onDragEnd` in `collectionTreeView` — I'm saying the code that shows the prompt shouldn't be decoupled from the code that actually blocks the drop, which is in `canDropCheck` in `collectionTreeView`. It doesn't make sense to block in one place and show a prompt to the user in another place.

The only advantage to triggering it from the item side (ignoring the blocking part) is that it could work when you have multiple Zotero panes open and drag from one items list to the other, but 1) that's only possible in Zotero for Firefox (and might not be possible in the future), 2) it doesn't actually work at the moment because we disallow cross-library cross-window drag (rather than duplicate all the `collectionTreeView` logic), and 3) it's possible you could drag a `zotero/item` from somewhere other than an `itemTreeView`, so it makes more sense to handle this from the drop code. If we do re-allow cross-window cross-library drag, we'll need to move the `canDropCheck` code to `libraryTreeView`, but the prompt should still stay with the blocking code, and right now that's in `collectionTreeView`